src/main/java/io/github/opencubicchunks/cubicchunks/chunk/cube/CubeWorldRegionColumn.java

```diff
@@ -0,0 +1,351 @@
+package io.github.opencubicchunks.cubicchunks.chunk.cube;
+
+import static io.github.opencubicchunks.cubicchunks.utils.Coords.*;
+
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import com.google.common.collect.Maps;
+import io.github.opencubicchunks.cubicchunks.CubicChunks;
+import io.github.opencubicchunks.cubicchunks.chunk.IBigCube;
+import io.github.opencubicchunks.cubicchunks.chunk.biome.ProtoColumnBiomeContainer;
+import io.github.opencubicchunks.cubicchunks.utils.Coords;
+import io.github.opencubicchunks.cubicchunks.world.CubeWorldGenRegion;
+import io.github.opencubicchunks.cubicchunks.world.DummyHeightmap;
+import it.unimi.dsi.fastutil.longs.LongSet;
+import it.unimi.dsi.fastutil.shorts.ShortList;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.chunk.ChunkBiomeContainer;
+import net.minecraft.world.level.chunk.ChunkStatus;
+import net.minecraft.world.level.chunk.LevelChunkSection;
+import net.minecraft.world.level.chunk.ProtoChunk;
+import net.minecraft.world.level.chunk.ProtoTickList;
+import net.minecraft.world.level.chunk.UpgradeData;
+import net.minecraft.world.level.levelgen.GenerationStep;
+import net.minecraft.world.level.levelgen.Heightmap;
+import net.minecraft.world.level.levelgen.feature.StructureFeature;
+import net.minecraft.world.level.levelgen.structure.StructureStart;
+import net.minecraft.world.level.lighting.LevelLightEngine;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.level.material.FluidState;
+import org.jetbrains.annotations.Nullable;
+
+//TODO: Implement this properly for mods. Vanilla is fine.
+public class CubeWorldRegionColumn extends ProtoChunk {
+
+    private final int xSectionOffset;
+    private final int zSectionOffset;
+    private final IBigCube mainCube;
+    private final IBigCube[] delegates;
+    private final CubeWorldGenRegion cubeWorldGenRegion;
+    private final LevelChunkSection[] sections;
+    private final Map<Heightmap.Types, Heightmap> heightmaps;
+    private final ProtoColumnBiomeContainer protoColumnBiomeContainer;
+
+
+    public CubeWorldRegionColumn(ChunkPos chunkPos, int xSectionOffset, int zSectionOffset, UpgradeData upgradeData, IBigCube[] delegates, CubeWorldGenRegion cubeWorldGenRegion) {
+        super(chunkPos, upgradeData, cubeWorldGenRegion);
+        this.xSectionOffset = xSectionOffset;
+        this.zSectionOffset = zSectionOffset;
+
+        this.mainCube = delegates[delegates.length / 2];
```

Comment:
Does this cube have any special purpose or is it simply the one in the middle? In that case `centerCube` would be a more descriptive name.